### ISSUE TITLE
v4.3 OLM setup - issue1: svc

### ISSUE THREAD

**Issue by tirelibirefe** [CrunchyData/postgres-operator#1480]
**Describe the bug**

Setup via OLM doesn't create svc

**To Reproduce**

'I downloaded and edited https://operatorhub.io/install/postgresql.yaml'

```
```
`apiVersion: v1
kind: Namespace
metadata:
  name: pgo
---
apiVersion: operators.coreos.com/v1
kind: OperatorGroup
metadata:
  name: operatorgroup
  namespace: pgo
spec:
  targetNamespaces:
  - pgo
---
apiVersion: operators.coreos.com/v1alpha1
kind: Subscription
metadata:
  name: pgo
  namespace: pgo
spec:
  channel: stable
  name: postgresql
  source: operatorhubio-catalog
  sourceNamespace: olm`´
```

```
$PGOROOT/conf/postgres-operator/pgo.yaml

PrimaryStorage: vmware-storage
ReplicaStorage: vmware-storage
BackrestStorage: vmware-storage

vmware-storage:
  AccessMode:  ReadWriteOnce
  Size:  1G
  StorageType:  dynamic
  StorageClass:  sas
  Fsgroup:  26

cd $PGOROOT/deploy
$PGOROOT/deploy/gen-api-keys.sh

kubectl create namespace pgo

```
kubectl create secret generic -n pgo pgo-backrest-repo-config \
  --from-file=config=$PGOROOT/conf/pgo-backrest-repo/config \
  --from-file=sshd_config=$PGOROOT/conf/pgo-backrest-repo/sshd_config
```

```
kubectl create secret generic -n pgo pgo-auth-secret \
  --from-file=server.crt=$PGOROOT/conf/postgres-operator/server.crt \
  --from-file=server.key=$PGOROOT/conf/postgres-operator/server.key
```

vi $PGOROOT/deploy/install-bootstrap-creds.sh # changed credentials

`$PGOROOT/deploy/install-bootstrap-creds.sh`

replaced $PGO_CMD with kubectl at line 32 and 39. The code $PGO_CMD should be changed or documentation should be updated. That was mentioned before as an issue.

```
kubectl create secret -n pgo tls pgo.tls \
  --key=$PGOROOT/conf/postgres-operator/server.key \
  --cert=$PGOROOT/conf/postgres-operator/server.crt
```
  
```
kubectl create configmap -n pgo pgo-config \
  --from-file=$PGOROOT/conf/postgres-operator
```

`kubectl create -f postgresql.yaml`

**Please tell us about your environment:**

* Operating System:
* Where is this running ( Local, Cloud Provider)

`bare-metal Kubernetes 1.18 on vSphere`

* Storage being used (NFS, Hostpath, Gluster, etc):

`vSphere CNS`

```
`[tireli@vdi-mk4-cnt ~]$ kubectl get pods -n pgo
NAME                                 READY   STATUS    RESTARTS   AGE
postgres-operator-775498f9f9-btcd2   4/4     Running   2          59m
[tireli@vdi-mk4-cnt ~]$ kubectl get csv -n pgo
NAME                      DISPLAY                             VERSION   REPLACES   PHASE
postgresoperator.v4.2.2   Crunchy PostgreSQL for Kubernetes   4.2.2                Succeeded
[tireli@vdi-mk4-cnt ~]$ kubectl get events -n pgo
LAST SEEN   TYPE      REASON                OBJECT                                          MESSAGE
<unknown>   Normal    Scheduled             pod/postgres-operator-775498f9f9-btcd2          Successfully assigned pgo/postgres-operator-775498f9f9-btcd2 to kik-worker2
59m         Normal    Pulling               pod/postgres-operator-775498f9f9-btcd2          Pulling image "crunchydata/pgo-apiserver:centos7-4.2.2"
58m         Normal    Pulled                pod/postgres-operator-775498f9f9-btcd2          Successfully pulled image "crunchydata/pgo-apiserver:centos7-4.2.2"
58m         Normal    Created               pod/postgres-operator-775498f9f9-btcd2          Created container apiserver
58m         Normal    Started               pod/postgres-operator-775498f9f9-btcd2          Started container apiserver
58m         Normal    Pulling               pod/postgres-operator-775498f9f9-btcd2          Pulling image "crunchydata/postgres-operator:centos7-4.2.2"
58m         Normal    Pulled                pod/postgres-operator-775498f9f9-btcd2          Successfully pulled image "crunchydata/postgres-operator:centos7-4.2.2"
58m         Normal    Created               pod/postgres-operator-775498f9f9-btcd2          Created container operator
58m         Normal    Started               pod/postgres-operator-775498f9f9-btcd2          Started container operator
58m         Normal    Pulling               pod/postgres-operator-775498f9f9-btcd2          Pulling image "crunchydata/pgo-scheduler:centos7-4.2.2"
58m         Normal    Pulled                pod/postgres-operator-775498f9f9-btcd2          Successfully pulled image "crunchydata/pgo-scheduler:centos7-4.2.2"
58m         Normal    Created               pod/postgres-operator-775498f9f9-btcd2          Created container scheduler
58m         Normal    Started               pod/postgres-operator-775498f9f9-btcd2          Started container scheduler
58m         Normal    Pulling               pod/postgres-operator-775498f9f9-btcd2          Pulling image "crunchydata/pgo-event:centos7-4.2.2"
58m         Normal    Pulled                pod/postgres-operator-775498f9f9-btcd2          Successfully pulled image "crunchydata/pgo-event:centos7-4.2.2"
58m         Normal    Created               pod/postgres-operator-775498f9f9-btcd2          Created container event
58m         Normal    Started               pod/postgres-operator-775498f9f9-btcd2          Started container event
58m         Normal    Pulled                pod/postgres-operator-775498f9f9-btcd2          Container image "crunchydata/pgo-apiserver:centos7-4.2.2" already present on machine
58m         Warning   Unhealthy             pod/postgres-operator-775498f9f9-btcd2          Liveness probe failed: find: '/tmp/scheduler.hb': No such file or directory
58m         Normal    Killing               pod/postgres-operator-775498f9f9-btcd2          Container scheduler failed liveness probe, will be restarted
58m         Normal    Pulled                pod/postgres-operator-775498f9f9-btcd2          Container image "crunchydata/pgo-scheduler:centos7-4.2.2" already present on machine
59m         Normal    SuccessfulCreate      replicaset/postgres-operator-775498f9f9         Created pod: postgres-operator-775498f9f9-btcd2
59m         Normal    ScalingReplicaSet     deployment/postgres-operator                    Scaled up replica set postgres-operator-775498f9f9 to 1
59m         Normal    RequirementsUnknown   clusterserviceversion/postgresoperator.v4.2.2   requirements not yet checked
59m         Normal    RequirementsNotMet    clusterserviceversion/postgresoperator.v4.2.2   one or more requirements couldn't be found
59m         Normal    AllRequirementsMet    clusterserviceversion/postgresoperator.v4.2.2   all requirements found, attempting install
59m         Normal    InstallSucceeded      clusterserviceversion/postgresoperator.v4.2.2   waiting for install components to report healthy
59m         Normal    InstallWaiting        clusterserviceversion/postgresoperator.v4.2.2   installing: Waiting: waiting for deployment postgres-operator to become ready: Waiting for rollout to finish: 0 of 1 updated replicas are available...
58m         Normal    InstallSucceeded      clusterserviceversion/postgresoperator.v4.2.2   install strategy completed with no errors
[tireli@vdi-mk4-cnt ~]$ kubectl get svc -n pgo
No resources found in pgo namespace.
[tireli@vdi-mk4-cnt ~]$`
```

**Comment by cbandy** [comment]
> Setup via OLM doesn't create svc

This is true. OLM doesn't provide a mechanism for creating Services at this time.

The instructions included with the OLM package show one way to create this service in the "After You Install" section:

> ```
> kubectl expose deployment -n pgo postgres-operator --type=LoadBalancer
> ```

We haven't published an OLM package for 4.3 yet, but the instructions coming in #1522 include this:

> The client needs to be able to reach the PostgreSQL Operator API from outside the Kubernetes cluster. Create an external service or forward a port locally.
> 
> ```
> kubectl -n "$PGO_OPERATOR_NAMESPACE" expose deployment postgres-operator --type=LoadBalancer
> 
> export PGO_APISERVER_URL="https://$(
>   kubectl -n "$PGO_OPERATOR_NAMESPACE" get service postgres-operator \
>     -o jsonpath="{.status.loadBalancer.ingress[*]['ip','hostname']}"
> ):8443"
> ```
> _or_
> ```
> kubectl -n "$PGO_OPERATOR_NAMESPACE" port-forward deployment/postgres-operator 8443
> 
> export PGO_APISERVER_URL="https://127.0.0.1:8443"
> ```

**Comment by cbandy** [comment]
OperatorHub.io packages have been merged:

* [operator-framework/community-operators#1832](https://github.com/operator-framework/community-operators/pull/1832)
* [operator-framework/community-operators#1833](https://github.com/operator-framework/community-operators/pull/1833)
